File: c/recursion/tower_of_hanoi.c
#include "stdio.h"
#include "stdlib.h"

int count = 0;

void tower_of_hanoi(int size, char source, char aux, char dest)
{
    count++;

    if (size == 1)
        printf("move o disco 1 da haste %c para %c\n", source, dest);

    else
    {
        tower_of_hanoi(size - 1, source, dest, aux);
        printf("move o disco %d da haste %c para %c\n", size, source, dest);

        tower_of_hanoi(size - 1, aux, source, dest);
    }
}

int main(int argc, char const *argv[])
{
   
    tower_of_hanoi(19, 'S', 'A', 'D');
    printf("\nnumero de movimentos %d\n\n", count);
    return 0;
}

File: c/recursion/tower_of_hanoi.py

count = {
    "a": 0
}

def tower_of_hanoi( size,  source, aux,  dest):
    count['a'] += 1

    if (size == 1):
        print(f"move o disco 1 da haste {source} para {dest}")

    else:
        tower_of_hanoi(size - 1, source, dest, aux)
        print(f"move o disco {size} da haste {source} para {dest}")

        tower_of_hanoi(size - 1, aux, source, dest)


tower_of_hanoi(19, 'S', 'A', 'D')
print(f"\nnumero de movimentos {count['a']}\n\n" )
